tests/FunctionsTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+use \PHPUnit\Framework\TestCase;
+
+class FunctionsTest extends TestCase
+{
+    public function setUp()
+    {
+    }
+
+    private function defaultPhonebook(): SimpleXMLElement
+    {
+        $xml =                 <<<EOD
+<?xml version="1.0" encoding="UTF-8"?>
+<phonebooks>
+    <phonebook name="Telefonbuch">
+        <contact>
+            <carddav_uid>ABCDEFGH-8AA4-4389-A2BE-18A42A61D24D</carddav_uid>
+            <telephony>
+                <number id="0" type="work">09131 123456</number>
+                <number id="1" type="home">0911 123456</number>
+                <number id="2" type="mobile">0152123456</number>
+            </telephony>
+            <person>
+                <realName>Mustermann, Max</realName>
+                <imageURL>file:///var/InternerSpeicher/MyUSB/FRITZ/fonpix/ABCDEFGH-8AA4-4389-A2BE-18A42A61D24D.jpg</imageURL>
+            </person>
+        </contact>
+        <contact>
+            <carddav_uid>ABCDEFGH-8AA4-4389-A2BE-18A42A61D24E</carddav_uid>
+            <telephony>
+                <number id="0" type="work">09131 345678</number>
+                <number id="1" type="home">0911 345678</number>
+                <number id="2" type="mobile">0152345678</number>
+            </telephony>
+            <person>
+                <realName>Mustermann, Marianne</realName>
+                <imageURL>file:///var/InternerSpeicher/MyUSB/FRITZ/fonpix/ABCDEFGH-8AA4-4389-A2BE-18A42A61D24E.jpg</imageURL>
+            </person>
+        </contact>
+    </phonebook>
+</phonebooks>
+EOD;
+
+        return simplexml_load_string($xml);
+    }
+
+    private function injectQuickDialAndVanity(SimpleXMLElement $phonebook, $contIndex, $numIndex, $qd, $van): SimpleXMLElement
+    {
+        $phonebook->phonebook->contact[$contIndex]->telephony->number[$numIndex]['quickdial'] = $qd;
+        $phonebook->phonebook->contact[$contIndex]->telephony->number[$numIndex]['vanity'] = $van;
+        return $phonebook;
+    }
+
+
+//                              _______        _      _____
```

Comment:
remove ;)

Reply:
:-( This helped me to navigate... OK - you'da boss.